src/server/data-access/MwBotWikibaseMessagesRepo.ts

```diff
@@ -0,0 +1,86 @@
+import TechnicalProblem from '@/common/data-access/error/TechnicalProblem';
+import { MessageKeys } from '@/common/MessageKeys';
+import MessageTranslationCollection from '@/datamodel/MessageTranslationCollection';
+import MessageCollection from '@/datamodel/MessageCollection';
+import MessagesRepository from '@/common/data-access/MessagesRepository';
+import mwbot from 'mwbot';
+
+interface RespondMessage {
+	[ key: string]: string;
+}
+
+export default class MwBotWikibaseMessagesRepo implements MessagesRepository {
+	private bot: mwbot;
+	private messageKeys: string;
+
+	public constructor( bot: mwbot, messageKeys?: string[] ) {
+		this.bot = bot;
+		if ( messageKeys === null || typeof messageKeys === 'undefined' ) {
+			this.messageKeys = this.getRequestedKeys();
+		} else {
+			this.messageKeys = messageKeys.join( '|' );
+		}
+	}
+
+	public getMessagesInLanguage( inLanguage: string|null ): Promise<MessageTranslationCollection> {
```

Comment:
I did not try to figure out which version of which branch this change is following but the usefulness of invoking a method called `*InLanguage` without a language parameter escapes me. The parameter should be required.
If (speculating) another method, implementing that same interface, decides not to make use of that parameter, that is its (lonesome) decision.

Reply:
right...and we can trust the mw fallback chain...